modules/nf-commons/src/main/nextflow/plugin/PluginsFacade.groovy

```diff
@@ -185,16 +185,20 @@ class PluginsFacade implements PluginStateListener {
 
     PluginManager getManager() { manager }
 
+    // Fix for plugins issue observed
     void init(boolean embedded=false) {
-        if( manager )
-            throw new IllegalArgumentException("Plugin system already setup")
```

Comment:
Based on the linked issue, they are using the `K8sDriverLauncher` to launch Nextflow jobs from a Spring Boot app and that causes Nextflow to try to create the plugins manager multiple times

Reply:
Thanks Ben, that's correct. So the change I propose here is to reuse the PluginManager that was already created instead of throwing the IllegalArgumentException.

Reply:
But the bigger issue here is that `init` is being invoked more than once which is breaking some core assumption. I would like to understand why using the K8s driver launcher as a library is breaking this assumption, as it could cause other issues down the line.

Have you tested your application with this PR? Does it work well?

Reply:
Yes, I applied this change on a fork and our team internally tested it on our systems. Works fine. No issues were encountered.

Reply:
Sorry, I'm nit inclined to merge this. The exception is thrown on purpose to guarantee the `init` method is invoked exactly one time.

Reply:
@pditommaso  @bentsherman 
the code fix I am proposing here does not call the init method more than once. In fact it works with the same initialized plugin manager created as a part of the first call to init.
 
The sequence of calls go this way -> `K8SDriverLauncher.run()` -> `makeConfig()` -> `loadConfig()` -> `Plugins.init()` -> `PluginsFacade.init()` 

We are using the K8SDriverLauncher to launch workflow pods programmatically (by calling `K8SDriverLauncher.run()`). Every workflow is run in its pod so we need to call the run method for every workflow we launch.

The problem with throwing an exception in the `PluginsFacade.init()` is that when I launch workflow 1 (by calling `K8SDriverLauncher.run()`) it runs well but when I launch workflow 2 it encounters the exception which does not allow my code to launch workflow pods in succession. Every call to `K8SDriverLauncher.run()` calls the `PluginsFacade.init()`. 

Either I am not using the `K8SDriverLauncher.run()` correctly or `PluginsFacade.init()` should not be called from within the `K8SDriverLauncher.run()` flow like the way it is doing now. 
 
Also, it is unclear to me why throw an exception,  why not re-use the plugins manager that already exists (which is what my fix does)?

Reply:
I see, so you are hitting this issue by launching multiple workflows through the K8s driver launcher

Since the K8s driver launcher is not really designed to support library usage, even if it happens to almost work, I think it would be better for you to implement your own version of the K8s driver launcher in your application that is designed from the ground up to support your use case, even if that means duplicating some code from Nextflow. You should be able to do that without maintaining a fork of Nextflow